inc/agent.class.php

```diff
@@ -1116,7 +1116,8 @@ protected function enrollByInvitationToken($input) {
       $agentAccount = new User();
       $agentAccount->add([
          'usercategories_id' => $config['agentusercategories_id'],
-         'name'              => $serial,
+         'name'              => 'flyvemdm-' . uniqid(),
```

Comment:
OK for the idea, but we need to check that uniqid() is "strong" enough to avoid collisions over time. 

Also this has impact on the credentials used by he agent to consume  the REST API. As the login is not predictible from the point of view of the agent, we need to enhance the invitation process because the agents use serial as login and a password provided in the invitation data to consule the REST API and enroll themselves.

Reply:
OK, I will add some logic to be more unique. I have talk with @rafaelje about this and the other part of the solution would be to return the agent account name to the device and store it to other uses.

Reply:
Have a look on this: https://stackoverflow.com/questions/1155008/how-unique-is-uuid

Also, my badn the enrollment only uses a user token so this PR will be kelt simple (but I still have some ideas :) )

Reply:
I think that using a UUID generator would be a good option. Here is one here http://php.net/manual/en/function.uniqid.php#94959

We might use UUID version 3 or 5 for username, and define their namespace with a V4 UUID.